src/map.rs

```diff
@@ -265,24 +277,47 @@ impl Map {
                     let green_mesh = graphics::Mesh::new_rectangle(
                         ctx, graphics::DrawMode::fill(), green_rect, graphics::Color::new(0., 1., 0., 1.)
                     ).unwrap();
-                    graphics::draw(ctx, &red_mesh, graphics::DrawParam::default()).unwrap();
-                    graphics::draw(ctx, &green_mesh, graphics::DrawParam::default()).unwrap();
+                    graphics::draw(ctx, &red_mesh, (na::Point2::new(0., 0.) + hit_offset,))
+                        .unwrap();
+                    graphics::draw(ctx, &green_mesh, (na::Point2::new(0., 0.) + hit_offset,))
+                        .unwrap();
 
                 }
             }
         }
 
+        Self::draw_red_hit_effect(hit_effect_timer, ctx);
+
         graphics::draw(ctx, &bullet_sb, (na::Point2::new(0., 0.),)).unwrap();
         if let Some(my_player) = game_state.get_player_by_id(my_id) {
             Map::draw_mini_map(game_state, &mut miniplane_sb, ctx, &my_player);
         }
 
         graphics::draw_queued_text(
-            ctx, (na::Point2::new(0., 0.),), None, graphics::FilterMode::Linear);
-
+            ctx, (na::Point2::new(0., 0.) + hit_offset,), None, graphics::FilterMode::Linear);
+        
         Self::draw_killfeed(ctx, assets, game_state);
     }
 
+    fn draw_red_hit_effect(hit_effect_timer: f32, ctx: &mut ggez::Context) {
+        if hit_effect_timer > 0. {
+            let rect = &graphics::Mesh::new_rectangle(
+                ctx,
+                graphics::DrawMode::fill(),
+                graphics::Rect::new(
+                    -constants::WINDOW_SIZE/2.,
+                    -constants::WINDOW_SIZE/2.,
+                    constants::WINDOW_SIZE,
+                    constants::WINDOW_SIZE,
+                    ),
+                [0.8, 0., 0., hit_effect_timer].into()
```

Comment:
Changing the hit effect timer will make the screen reder, I would prefer having `hit_effect_timer / MAX_HIT_EFFECT_TIMER * HIT_EFFECT_MAX_ALPHA`

Reply:
Fixed